CyclopsNuclearReactor/CyNukReactorSMLHelper.cs

```diff
@@ -0,0 +1,211 @@
+
+        private const string NoPowerKey = "CyNukeNoPower";
+        public static string NoPoweMessage()
+        {
+            return Language.main.Get(NoPowerKey);
+        }
+
+        public static TechType TechTypeID { get; private set; }
+
+        public static void PatchSMLHelper()
+        {
+
+            if (!main.GetPrefabs())
+            {
+                throw new NullReferenceException("CyNukReactor has failed to retrieve the prefab from the asset bundle");
+            }
+
+            main.Patch();
+        }
+
+        public CyNukReactorSMLHelper() : base("CyNukReactor", "Cyclops Nuclear Reactor", "A nuclear reactor re-designed to fit and function inside the Cyclops.")
+        {
+            OnFinishedPatching += AdditionalPatching;
+        }
+
+        public override TechGroup GroupForPDA { get; } = TechGroup.InteriorModules;
+        public override TechCategory CategoryForPDA { get; } = TechCategory.InteriorModule;
+        public override string AssetsFolder { get; } = "CyclopsNuclearReactor/Assets";
+
+        public override GameObject GetGameObject()
+        {
+            SubRoot cyclops = Player.main.currentSub;
+            if (cyclops != null)
+            {
+                var mgr = CyNukeChargeManager.GetManager(cyclops);
+                mgr.SyncReactorsExternally();
+
+                if (mgr.CyNukeReactors.Count >= CyNukeChargeManager.MaxReactors)
+                {
+                    ErrorMessage.AddMessage(OverLimitMessage());
+                    return null;
+                }
+            }
+
+            var prefab = GameObject.Instantiate(_cyNukReactorPrefab);
+            GameObject consoleModel = prefab.FindChild("model");
+
+            // Update sky applier
+            SkyApplier skyApplier = prefab.AddComponent<SkyApplier>();
+            skyApplier.renderers = consoleModel.GetComponentsInChildren<MeshRenderer>();
+            skyApplier.anchorSky = Skies.Auto;
+
+            //Add the constructible component to the prefab
+            Constructable constructible = prefab.AddComponent<Constructable>();
+
+            constructible.allowedInBase = false;
+            constructible.allowedInSub = true; // Only allowed in Cyclops
+            constructible.allowedOutside = false;
+            constructible.allowedOnCeiling = false;
+            constructible.allowedOnGround = true; // Only on ground
+            constructible.allowedOnWall = false;
+            constructible.allowedOnConstructables = false;
+            constructible.controlModelState = true;
+            constructible.rotationEnabled = true;
+            constructible.techType = this.TechType;
+            constructible.model = consoleModel;
+
+            //Add the prefabIdentifier
+            prefab.AddComponent<PrefabIdentifier>().ClassId = this.ClassID;
+
+            //Add the techType to this custom prefab
+            TechTag techTag = prefab.AddComponent<TechTag>();
+            techTag.type = this.TechType;
```

Comment:
This isn't needed.  
`PrefabIdentifier` and `TechTag` are handled by the `ModPrefab` base class (`Buildable` inherits from `ModPrefab`).

This could be the reason why the `Id` field of `PrefabIdentifier` is returning null.